modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/ClickGoalPicker/ClickGoalPicker.es.js

```diff
@@ -376,9 +378,11 @@ function OverlayContainer({allowEdit, root}) {
 
 	const handleMouseUp = useCallback(
 		(event) => {
-			if (mousedownRef.current === true) {
-				event.preventDefault();
-				stopImmediatePropagation(event);
+			const overlayTarget = event.target.closest(
+				`.${OVERLAY_TARGET_CLASS}`
+			);
+
+			if (mousedownRef.current === true && !overlayTarget) {
```

Comment:
The issue here was that we were calling `deactivate` when the user was clicking on an overlay target. By adding this logic, we make sure to not deactivate if its an overlay target. You should not be able to select a new target and also remove an existing target